types/types.go

```diff
@@ -992,6 +994,23 @@ func CreateDisksFromAnnotation(annotation string) (map[string]longhorn.DiskSpec,
 		if disk.StorageReserved < 0 || disk.StorageReserved > diskStat.StorageMaximum {
 			return nil, fmt.Errorf("the storageReserved setting of disk %v is not valid, should be positive and no more than storageMaximum and storageAvailable", disk.Path)
 		}
+		if disk.StorageReserved == 0 {
+			if disk.Type == longhorn.DiskTypeBlock {
+				file, err := os.Open(ReplicaHostPrefix + disk.Path)
+				if err != nil {
+					return nil, fmt.Errorf("failed to open block device at %s: %w", disk.Path, err)
+				}
+				defer file.Close()
+				var size uint64
+				_, _, errno := unix.Syscall(unix.SYS_IOCTL, file.Fd(), 0x80081272, uintptr(unsafe.Pointer(&size)))
```

Comment:
Does the IOCTL command code (0x80081272) work for both SATA and NVMe SSDs?

Reply:
This has been tested in a SATA SSD environment only. I will verify whether it is supported on NVMe as well and will share the findings once confirmed

Reply:
Thanks @jangseon-ryu